Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py

```diff
@@ -225,21 +243,26 @@ def fetch_mails(client: IMAPClient,
                mails will be fetched (used with list-messages command)
         save_file: Whether to save the .eml file of the incident's mail
         message_id: A unique message ID with which a specific mail can be fetched
+        uid_to_fetch_from: The email message UID to start the fetch from as offset
 
     Returns:
         mails_fetched: A list of Email objects
-        messages: A list of the ids of the messages fetched
+        messages_fetched: A list of the ids of the messages fetched
+        last_message_in_current_batch: The UID of the last message fetchedd
     """
     if message_id:
         messages = [message_id]
     else:
         messages_query = generate_search_query(first_fetch_time,
                                                permitted_from_addresses,
-                                               permitted_from_domains)
+                                               permitted_from_domains,
+                                               uid_to_fetch_from)
         messages = client.search(messages_query)
         limit = len(messages) if limit == -1 else limit
```

Comment:
We shouldn't allow unlimited fetch. There should be a max enforcement of 200. If I understand the code correctly.